jdbc/src/java/org/apache/hive/jdbc/saml/HiveJdbcBrowserClient.java

```diff
@@ -200,10 +206,47 @@ private Map<String, String> getQueryParams(URI ssoUri)
     return paramMap;
   }
 
+  /**
+   * The pattern for valid urls (http/s).
+   */
+  private static final Pattern VALID_URL = createValidUrlPattern();
+  private static Pattern createValidUrlPattern() {
+    final String validURLPattern =
```

Comment:
The validation pattern is pretty complex and as Sonar points out it can lead to Security Hotspots etc. Do we need this check to be that complicated? What exactly do we want to achieve?

If the intention is to allow only redirection to HTTP, HTTPS, URIs then the check can be made much simpler by using `URI.getScheme` method. As a matter of fact the `JdbcBrowserClientContext` or `org.apache.hive.jdbc.saml.HiveJdbcSamlRedirectStrategy#createLocationURI` classes may be a better place to impose restrictions around the URI.

Reply:
I thought about pushing the control up but I did not want to disrupt current error handling. Let me try harder and see. As for the regex, it may be overkill but it tries hard to ensure nothing 'extraordinary' appears in that URL.

Reply:
The URL and URI constructors are already trying to avoid unexpected things in there. If we need to have more complex checks then we should have a concrete reason.

Reply:
New solution posted.